### PR TITLE
DEV-3064: extend js interface

### PR DESCRIPTION
**Pull request by ndillon1**
- Add TS interface for IMA, metadata, theme, translations and button configuration 
- Add TYPES TYPES TYPES
- Remove prop types

### CODE REVIEW

**Review by yaroslavlvov — APPROVED** [review]

File: types/buttons.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface IVideoPlayerButtons {
+  watchlist?: boolean;
+  favourite?: boolean;
+}
```
Comment (markhomoki) [inline types/buttons.ts]:
```suggestion
}

```

File: utils/ima.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      config.contentSourceId = matches[1];
+      config.videoId = matches[2];
+      config.authToken = matches[3].replace(/%3D/g, '=').replace(/%7E/g, '~');
+    } else if (!isVod && matches[1]) {
+      config.assetKey = matches[1];
+    }
+  }
+
+  return config;
+}
```
Comment (markhomoki) [inline utils/ima.ts]:
```suggestion
}

```

File: utils/ima.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { IVideoPlayerIMA } from "../types/ima";
```
Comment (markhomoki) [inline utils/ima.ts:1]: single quote

File: types/theme.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+
+export interface IVideoPlayerTheme {
+  colors: IVideoPlayerThemeObject,
+  fonts: IVideoPlayerThemeObject
+}
+
+
```
Comment (markhomoki) [inline types/theme.ts:7]:
```suggestion
```

File: types/theme.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (markhomoki) [inline types/theme.ts:1]:
```suggestion
```

File: types/source.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { IVideoPlayerDRM } from "./drm";
+import { IVideoPlayerIMA } from "./ima";
+import { IVideoPlayerMetadata } from "./metadata";
+import { IMuxData } from "./mux";
+import { IVideoPlayerSubtitles } from "./subtitles";
```
Comment (markhomoki) [inline types/source.ts]: single quote

File: types/player.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { ViewProps } from "react-native";
+
+import { IVideoPlayerButtons } from "./buttons";
+import { IVideoPlayerCallbacks } from "./callbacks";
+import { VideoResizeMode } from "./resizeMode";
+import { IVideoPlayerSource } from "./source";
```
Comment (markhomoki) [inline types/player.ts]: single quote

File: types/player.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  labelFontName?: string;
+  live: boolean;
+  overlayAutoHideTimeout: number,
+  playInBackground: boolean;
+  source: IVideoPlayerSource;
+  resizeMode: VideoResizeMode;
+  translations: IVideoPlayerTranslations;
+  stateMiddleCoreControls: string;
+}
+
```
Comment (markhomoki) [inline types/player.ts]:
```suggestion
```

**Review by markhomoki — APPROVED** [review]

**Pull request merged**